plugins/by-name/clangd-extensions/default.nix

```
@@ -76,18 +76,34 @@ lib.nixvim.plugins.mkNeovimPlugin {
 
   extraConfig = cfg: {
     warnings = lib.nixvim.mkWarnings "plugins.clangd-extensions" {
-      when = !config.plugins.lsp.enable;
+      when =
+        !(
+          (config.plugins.lsp.servers.clangd.enable or false) || (config.lsp.servers.clangd.enable or false)
+        );
 
       message = ''
-        You have enabled `clangd-extensions` but not the lsp (`plugins.lsp`).
-        You should set `plugins.lsp.enable = true` to make use of the clangd-extensions' features.
+        You have enabled `clangd-extensions` but not the clangd lsp
+        (`plugins.lsp.servers.clangd` or `lsp.servers.clangd`). You should set
+        `plugins.lsp.servers.clangd.enable = true` or
+        `lsp.servers.clangd.enable = true` to make use of the
+        clangd-extensions' features.
       '';
     };
 
-    plugins.lsp = {
+    lsp = {
       servers.clangd = {
         enable = lib.mkDefault true;
+        settings = lib.mkIf cfg.enableOffsetEncodingWorkaround {
+          capabilities = {
+            offsetEncoding = [ "utf-16" ];
+          };
+        };
+      };
+    };
 
+    plugins.lsp = {
+      servers.clangd = {
+        enable = lib.mkIf (!config.lsp.clangd.enable or false) (lib.mkDefault true);
```

Nit: we shouldn't need `or false` here because there is an explicitly declared `config.lsp.servers.clangd.enable` option.

Typo: you missed `servers` from the attr path:

```suggestion
        enable = lib.mkIf (!config.lsp.servers.clangd.enable) (lib.mkDefault true);
```